python/kwiver/sprokit/adapters/adapter_data_set.cxx

```diff
@@ -0,0 +1,296 @@
+      py::arg("port"), py::arg("val").none(false))
+    .def("_add_uchar_vector", &adapter_data_set::add_value<std::shared_ptr<std::vector<unsigned char>>>,
+      py::arg("port"), py::arg("val").none(false))
+    // Next kwiver vital types
+    .def("_add_bounding_box", &adapter_data_set::add_value<kwiver::vital::bounding_box_d>)
+    .def("_add_timestamp", &adapter_data_set::add_value<kwiver::vital::timestamp>)
+    .def("_add_corner_points", &adapter_data_set::add_value<kwiver::vital::geo_polygon>)
+    .def("_add_f2f_homography", &adapter_data_set::add_value<kwiver::vital::f2f_homography>)
+
+    .def("empty", &adapter_data_set::empty)
+
+    // get_port_data is also templated
+    .def("_get_port_data_int", &adapter_data_set::get_port_data<int>)
+    .def("_get_port_data_float", &adapter_data_set::get_port_data<float>)
+    .def("_get_port_data_string", &adapter_data_set::get_port_data<std::string>)
+    // Next shared ptrs to kwiver vital types
+    .def("_get_port_data_image_container", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::image_container > >)
+    .def("_get_port_data_descriptor_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::descriptor_set > >)
+    .def("_get_port_data_detected_object_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::detected_object_set > >)
+    .def("_get_port_data_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::track_set > >)
+    .def("_get_port_data_feature_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::feature_track_set > >)
+    .def("_get_port_data_object_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::object_track_set > >)
+    //Next shared ptrs to native C++ types
+    .def("_get_port_data_double_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<double>>>)
+    .def("_get_port_data_string_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<std::string>>>)
+    .def("_get_port_data_uchar_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<unsigned char>>>)
+    //Next kwiver vital types
+    .def("_get_port_data_bounding_box", &adapter_data_set::get_port_data<kwiver::vital::bounding_box_d>)
+    .def("_get_port_data_timestamp", &adapter_data_set::get_port_data<kwiver::vital::timestamp>)
+    .def("_get_port_data_corner_points", &adapter_data_set::get_port_data<kwiver::vital::geo_polygon>)
+    .def("_get_port_data_f2f_homography", &adapter_data_set::get_port_data<kwiver::vital::f2f_homography>)
+    .def("__nice__", [](const adapter_data_set& self) -> std::string {
+    auto locals = py::dict(py::arg("self")=self);
+      py::exec(R"(
+          retval = 'size={}'.format(len(self))
+      )", py::globals(), locals);
+      return locals["retval"].cast<std::string>();
+      })
+  .def("__repr__", [](py::object& self) -> std::string {
```

Comment:
Doesn't have to happen here, but it might be useful to make a separate pybind11 class that other pybind11 classes can inherit from similar to [NiceRepr](https://ubelt.readthedocs.io/en/0.9.1/_modules/ubelt/util_mixins.html#NiceRepr) in ubelt. (although note that in this case the `__str__` method will still exist along with `__nice__`.